cocotb/drivers/amba.py

```diff
@@ -281,19 +275,18 @@ def write(self, address, value, size=None, burst=AXIBurst.INCR,
         write_data = self._send_write_data(value, data_latency, byte_enable,
                                            sync)
 
-        yield Combine(write_address, write_data)
+        await Combine(write_address, write_data)
```

Comment:
You can change this to
```suggestion
        await Combine(cocotb.fork(write_address), cocotb.fork(write_data))
```
which means you can drop the `coroutine`

Reply:
Isn't it going to break existing user code? E.g., in my test I have sometimes:
```
result = await First(axim.write(...), ClockCycles(...))
```
to check for a timeout in the write, however dropping `coroutine` on `write()` before #1846 is going to break it.

Reply:
I hadn't considered that, good point. That probably means I need to be more careful in #2022...

Reply:
For internal coroutines I'd still encourage you to remove the `@coroutine` in the way I suggest above, but I think you're right that the public API needs to remain decorated for now.